### PR TITLE
Fixed issue 18459: When filtering the survey list by a parent survey group, surveys assigned to a child survey group are not listed

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/models/Survey.php [hunk 1]
```diff
@@ -1624,7 +1623,23 @@ public function search()
 
         // Survey group filter
         if (isset($this->gsid)) {
-            $criteria->compare("t.gsid", $this->gsid, false);
+            // The survey group filter (from the dropdown, not by title search) is applied to five levels of survey groups.
+            // That is, it matches the group the survey is in, the parent group of that group, and the "grandparent" group, etc.
+            $groupJoins = 'LEFT JOIN {{surveys_groups}} surveygroup ON t.gsid = surveygroup.gsid ';
```
Comment (Shnoulle) [inline application/models/Survey.php:1628]: Unsure on the performance , a select in array can not be better ?

**Review by Shnoulle — APPROVED** [review]

File: application/models/Survey.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/Survey.php:1628]: No real issue

File: application/models/Survey.php [hunk 2]
```diff
@@ -1615,7 +1615,6 @@ public function search()
         // Search filter
         $sid_reference = (Yii::app()->db->getDriverName() == 'pgsql' ? ' t.sid::varchar' : 't.sid');
         $aWithRelations[] = 'owner';
-        $aWithRelations[] = 'surveygroup';
```
Comment (Shnoulle) [inline application/models/Survey.php:1618]: Needed (test by @maziminke )

**Review by Shnoulle — COMMENTED** [review]
Test broken

See with https://github.com/LimeSurvey/LimeSurvey/blob/0c9c5c8efe51b5af785c34b059c75ce206d906ad/application/models/Survey.php#LL1646C18-L1646C18 too

File: application/models/Survey.php (hunk 2 above)
Comment (Shnoulle) [reply application/models/Survey.php:1618]: But need another alias `Syntax error or access violation: 1066 Not unique table/alias: 'surveygroup'.`

**Comment by gabrieljenik** [comment]
Ready for re-reviewal...

File: application/extensions/admin/survey/ListSurveysWidget/ListSurveysWidget.php [hunk 3]
```diff
@@ -35,7 +35,6 @@ public function run()
         }
 
         $this->model->active = null;
-        $this->model->gsid  = null;
```
Comment (Shnoulle) [inline application/extensions/admin/survey/ListSurveysWidget/ListSurveysWidget.php:38]: It's already set in cntroller ! \o/ great

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**